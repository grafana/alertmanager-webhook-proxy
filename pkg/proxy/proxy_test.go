package proxy

import (
	"testing"

	"net/http"
)

type test struct {
	target string
}

var tests = []test{
	test{"https://taylorswift.com"},
	test{"https://grafana.com"},
	test{"https://play.grafana.com"},
}

func TestNew(t *testing.T) {
	for _, test := range tests {
		p, err := New(test.target)

		if err != nil {
			t.Errorf("Error: %v", err)
		}

		req, _ := http.NewRequest("POST", "https://replace.me", nil)

		p.Director(req)

		got := req.URL.Scheme + "://" + req.URL.Host + req.URL.Path
		want := test.target + "/"

		if got != want {
			t.Errorf("got %q, wanted %q", got, want)
		}
	}
}
