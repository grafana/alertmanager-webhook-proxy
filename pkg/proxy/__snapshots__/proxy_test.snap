
[TestDecode - 1]
template.Data{
    Receiver: "webhook",
    Status:   "firing",
    Alerts:   {
        {
            Status:       "firing",
            Labels:       {"alertname":"Test", "dc":"eu-west-1", "instance":"localhost:9090", "job":"prometheus24"},
            Annotations:  {"description":"some description"},
            StartsAt:     time.Date(2018, time.August, 3, 9, 52, 26, 739266876, time.Location("")),
            EndsAt:       time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC),
            GeneratorURL: "http://example.com:9090/graph?g0.expr=go_memstats_alloc_bytes+%3E+0&g0.tab=1",
            Fingerprint:  "",
        },
    },
    GroupLabels:       {"alertname":"Test", "job":"prometheus24"},
    CommonLabels:      {"alertname":"Test", "dc":"eu-west-1", "instance":"localhost:9090", "job":"prometheus24"},
    CommonAnnotations: {"description":"some description"},
    ExternalURL:       "http://example.com:9093",
}
---
